File: app-backend/src/main/java/dev/andrei/app_backend/dto/preference/AttributeConceptDto.java
package dev.andrei.app_backend.dto.preference;

import java.util.UUID;

/**
 * A concept the user can express a preference for, merged with their current importance
 * (0 when they have never set one). Drives the preference editor.
 */
public record AttributeConceptDto(
        UUID conceptId,
        String slug,
        String displayName,
        String groupLabel,
        int sortOrder,
        int importance
) {}

File: appfrontend/app/src/main/java/dev/andrei/app_frontend/data/remote/dto/preference/AttributeConceptDto.kt
package dev.andrei.app_frontend.data.remote.dto.preference

/** A concept the user can rate, with their current importance (0 when never set). */
data class AttributeConceptDto(
    val conceptId: String,
    val slug: String,
    val displayName: String,
    val groupLabel: String?,
    val sortOrder: Int,
    val importance: Int
)
